### PR TITLE
add tests for `SVGSpriteBase`

### PR DESCRIPTION
**Pull request by Kreeg**

### CODE REVIEW

File: lib/svg-sprite/mode/base.js [hunk 1]
```diff
@@ -135,7 +135,7 @@ SVGSpriteBase.prototype._initData = function(data) {
                 renderConfig.dest = path.resolve(this.config.dest, this.config.example.dest);
             }
         } else if (this.config.example !== true) {
-            renderConfig = false;
+            renderConfig = false; // todo: fix bug with this branch on line 142 (boolean has no "dest" property)
```
Comment (XhmikosR) [inline lib/svg-sprite/mode/base.js:138]: Which branch are you referring to?

File: test/svg-sprite/mode/base.test.js [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+                prefix: '1',
+                sprite: '',
+                example: {
+                    dest: '.'
+                }
+            };
+            const { cls } = getClassAndInitFn();
+            const base = new cls(TEST_SPRITER, TEST_CONFIG, {}, '');
+            const TEST_SVG = 'svg';
+            const TEST_SPRITE = 'sprite-ae8eb96d.svg';
```
Comment (XhmikosR) [inline test/svg-sprite/mode/base.test.js:426]: I'd just test for the number of characters here, like `^sprite-\w{8}\.svg$`. I prefer not using regex when possible but I think it makes sense here.

Comment (XhmikosR) [reply test/svg-sprite/mode/base.test.js:426]: BTW, unsure why we use 8 characters. I was under the impression 7 was more common.

File: lib/svg-sprite/mode/base.js (hunk 1 above)
Comment (Kreeg) [reply lib/svg-sprite/mode/base.js:138]: @XhmikosR current `else if` branch (line 138)

Comment (XhmikosR) [reply lib/svg-sprite/mode/base.js:138]: Ah, I see. Indeed, good catch 👍

**Review by XhmikosR — APPROVED** [review]

**Pull request merged**